### PR TITLE
Add CLDT to admin domains

### LINKED ISSUE #2520: Add CDLT organisation to workshop Administrator options
CDLT = Collaborative Lesson Development Training.

This can be done through the OrganizationManager class.

### PR DESCRIPTION
**Pull request by elichad**
fixes #2520.

This will need a dedicated release in the next couple of weeks as we are not close to being ready with v4.3.

### CODE REVIEW

File: amy/autoemails/actions.py
Comment (pbanaszkiewicz) [inline amy/autoemails/actions.py:495]: I'm afraid these relate to Action classes, which are pickled and stored in Redis. We may not be able to update the already scheduled emails.

Does this matter for the already scheduled emails?

File: amy/workshops/models.py
Comment (pbanaszkiewicz) [inline amy/workshops/models.py:81]: Does the `clean_domain` in `OrganizationForm` work with this domain?

**Comment by pbanaszkiewicz** [comment]
@elichad
> This will need a dedicated release in the next couple of weeks as we are not close to being ready with v4.3.

Please create a new release (4.2.1?) and a release branch from `release/4.2.0` (or main if we don't have it). Eventually this code will have to be merged into both `develop` and `release/4.2.1`.

**Review by pbanaszkiewicz — COMMENTED** [review]
Left some minor concerns. Once they're cleared this can be merged.

File: amy/autoemails/actions.py
Comment (elichad) [reply amy/autoemails/actions.py:495]: It shouldn't matter for already scheduled, as this domain can't be selected as an administrator until after we release this PR.

Do we have to take any extra steps to update these classes in Redis during deployment?

File: amy/workshops/models.py
Comment (elichad) [reply amy/workshops/models.py:81]: Yes, the format is covered by this test: https://github.com/carpentries/amy/blob/develop/amy/fiscal/tests/test_organization.py#L56

**Comment by elichad** [comment]
Created [release/v4.2.2](https://github.com/carpentries/amy/tree/release/v4.2.2) from release/v4.2.1.

File: amy/autoemails/actions.py
Comment (pbanaszkiewicz) [reply amy/autoemails/actions.py:495]: Everything scheduling these actions after the release will use the new conditions.

**Review by pbanaszkiewicz — APPROVED** [review]

**Comment by elichad** [comment]
Thanks @pbanaszkiewicz! I should have time to release this on Wednesday afternoon, so I'll merge it then (unless you want to do the release yourself sooner, which would be fine too)

**Pull request merged**
